src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -2146,6 +2069,121 @@ function escapeInvisibleChars(enumValue: string): string {
 		.replace(/\r/g, '\\r');
 }
 
+/**
+ * Controls logic and rendering of the label next to each setting header.
+ * For example, the "Modified by" and "Overridden by" labels go here.
+ */
+class SettingsTreeMiscLabel {
+	private labelElement: HTMLElement;
+	private otherOverridesElement: HTMLElement;
+	private syncIgnoredElement: HTMLElement;
+	private defaultOverrideIndicatorElement: HTMLElement;
+	private defaultOverrideIndicatorLabel: SimpleIconLabel;
+
+	constructor(container: HTMLElement) {
+		this.labelElement = DOM.append(container, $('.misc-label'));
+		this.labelElement.style.display = 'inline';
+
+		this.otherOverridesElement = this.createOtherOverridesElement();
+		this.syncIgnoredElement = this.createSyncIgnoredElement();
+		const { element, label } = this.createDefaultOverrideIndicator();
+		this.defaultOverrideIndicatorElement = element;
+		this.defaultOverrideIndicatorLabel = label;
+	}
+
+	private createOtherOverridesElement(): HTMLElement {
+		const otherOverridesElement = $('span.setting-item-overrides');
+		return otherOverridesElement;
+	}
+
+	private createSyncIgnoredElement(): HTMLElement {
+		const syncIgnoredElement = $('span.setting-item-ignored');
+		const syncIgnoredLabel = new SimpleIconLabel(syncIgnoredElement);
+		syncIgnoredLabel.text = `$(sync-ignored) ${localize('extensionSyncIgnoredLabel', 'Sync: Ignored')}`;
+		syncIgnoredLabel.title = localize('syncIgnoredTitle', "Settings sync does not sync this setting");
+		return syncIgnoredElement;
+	}
+
+	private createDefaultOverrideIndicator(): { element: HTMLElement, label: SimpleIconLabel } {
+		const defaultOverrideIndicator = $('span.setting-item-default-overridden');
+		const defaultOverrideLabel = new SimpleIconLabel(defaultOverrideIndicator);
+		return { element: defaultOverrideIndicator, label: defaultOverrideLabel };
+	}
+
+	private render() {
+		const elementsToShow = [this.otherOverridesElement, this.syncIgnoredElement, this.defaultOverrideIndicatorElement].filter(element => {
+			return element.style.display !== 'none';
+		});
+
+		this.labelElement.innerText = '';
+		this.labelElement.style.display = 'none';
+		if (elementsToShow.length) {
+			this.labelElement.style.display = 'inline';
+			DOM.append(this.labelElement, $('span', undefined, '('));
+			for (let i = 0; i < elementsToShow.length - 1; i++) {
+				DOM.append(this.labelElement, elementsToShow[i]);
+				DOM.append(this.labelElement, $('span', undefined, ','));
```

Comment:
Tiny nit, but can you add like 4px padding to the right of the `,` to make the groups breathe a little?

<img width="386" alt="CleanShot 2022-02-02 at 15 50 07@2x" src="https://user-images.githubusercontent.com/35271042/152257271-7112d88d-6800-4be9-8fce-207a9e50a756.png">